src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -135,6 +140,29 @@ def init_nat_gateway_models(self) -> None:
         # for model_name, model_type in nat_gateway_models.items():
         #     setattr(self, model_name, model_type)
 
+    def init_pod_identity_models(self) -> None:
```

Comment:
question(non-blocking): is it possible to decorate it with `@property` and make it a lazy initialization? This could make the caller side easier to use.

For example:

```python
@property
def pod_identity_models(self):
    if self._pod_identity_models not None:
      return self._pod_identity_models
    # otherwise, initialize the model

# in caller side:
profile = self.pod_identity_models.ManagedClusterPodIdentityProfile()
```

But since I don't know how the whole thing works in the new implementation, I might miss some contexts here.

Reply:
yeah looks like we only expect to initialize once, I think it's better to rename the method to something like `def _init_pod_identity_models` to avoid referencing outside current class.

Reply:
Good point, I'll wrap the dictionary with the @property decorator and I suppose without setting a setter function, this would also help avoid accidentally setting values.

Reply:
Updated in commit [#9e1827e](https://github.com/Azure/azure-cli-extensions/pull/4243/commits/9e1827e8f5150a8f3e3884c433c9d9e35f668c3d).